test/Spec/MLabsPAB/Contract.hs

```diff
@@ -53,92 +53,99 @@ tests =
+    Right tx ->
+      let outTxId = encodeByteString $ fromBuiltin $ TxId.getTxId $ Tx.txId tx
+       in state.commandHistory
+            @?= map
+              (Text.replace "\n" " ")
+              [ [text|
+                cardano-cli query utxo
+                --address ${addr1}
+                --mainnet
+               |]
+              , [text|
+                cardano-cli transaction build --alonzo-era
+                --tx-in ${inTxId}#0
+                --tx-in-collateral ${inTxId}#0
+                --tx-out ${addr2}+1000
+                --change-address ${addr1}
+                --required-signer signing-keys/signing-key-${pkh1'}.skey
+                --mainnet --protocol-params-file ./protocol.json --out-file txs/tx-${outTxId}.raw
+              |]
+              , [text|
+                cardano-cli transaction sign
+                --tx-body-file txs/tx-${outTxId}.raw
+                --signing-key-file signing-keys/signing-key-${pkh1'}.skey
+                --out-file txs/tx-${outTxId}.signed
+              |]
+              ]
 
 multisigSupport :: Assertion
 multisigSupport = do
   let txOutRef = TxOutRef "e406b0cf676fc2b1a9edb0617f259ad025c20ea6f0333820aa7cef1bfe7302e5" 0
       txOut = TxOut (Ledger.pubKeyHashAddress pkh1) (Ada.lovelaceValueOf 1250) Nothing
       initState = def {utxos = [(txOutRef, txOut)]}
-      txId = encodeByteString $ fromBuiltin $ getTxId $ txOutRefId txOutRef
+      inTxId = encodeByteString $ fromBuiltin $ TxId.getTxId $ Tx.txOutRefId txOutRef
 
-      contract :: Contract Text (Endpoint "SendAda" ()) Text ()
+      contract :: Contract Text (Endpoint "SendAda" ()) Text Tx
       contract = do
         let constraints =
               Constraints.mustPayToPubKey pkh2 (Ada.lovelaceValueOf 1000)
                 <> Constraints.mustBeSignedBy pkh3
-        void $ submitTx constraints
+        submitTx constraints
 
       (result, state, _) = runContractPure contract initState
   -- Building and siging the tx includes both signing keys
-  result @?= Right ()
-  state.commandHistory
-    @?= map
-      (Text.replace "\n" " ")
-      [ [text|
-          cardano-cli query utxo
-          --address ${addr1}
-          --mainnet
-          |]
-      , [text|
-          cardano-cli transaction build --alonzo-era
-          --tx-in ${txId}#0
-          --tx-in-collateral ${txId}#0
-          --tx-out ${addr2}+1000
-          --change-address ${addr1}
-          --required-signer signing-keys/signing-key-${pkh1'}.skey
-          --required-signer signing-keys/signing-key-${pkh3'}.skey
-          --mainnet --protocol-params-file ./protocol.json --out-file tx.raw
-        |]
-      , [text| 
-          cardano-cli transaction sign
-          --tx-body-file tx.raw
-          --signing-key-file signing-keys/signing-key-${pkh1'}.skey
-          --signing-key-file signing-keys/signing-key-${pkh3'}.skey
-          --out-file tx.signed
-        |]
-      ]
+  case result of
```

Comment:
Seeing this little pattern of case result to get the filename a lot, perhaps a little helper like `assertWithTxName` or something? Could save an indent on the rest of the func